low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,500 @@
+
+            except (FileNotFoundError, OSError) as e:
+                logger.error(f"Failed to open the SSPL 'state.txt' file with an error '{e}',\
+                    Can't determine the SSPL state, So enclosure logs also being collected.")
+                sspl_state = 'active'
+
+            self.get_ipmi_sel_data()
+            self.get_config_data()
+            self.get_drives_smart_data()
+            if sspl_state == "active" and not include_encl:
+                self.get_enclosure_logs()
+
+            if os.path.exists(self.enclosure_log) and sspl_state == 'active' \
+                    and not include_encl:
+                enclosure_zip_file = "enclosure-wwn-{0}-logs-{1}.zip".format(
+                    self.enclosure_wwn, str(int(time.time())))
+                shutil.copy(self.enclosure_log,
+                            self.tmp_bundle_path+enclosure_zip_file)
+                logger.info("Enclosure Log File Location: %s" %
+                            self.tmp_bundle_path+enclosure_zip_file)
+
+
+        except (OSError, tarfile.TarError) as err:
+            msg = "Facing problem while creating sspl support bundle : %s" % err
+            self.__clear_tmp_files()
+            raise SupportBundleError(1, msg)
+        try:
+            # Process request
+            BundleRequestHandler._update_request_status(
+                request_id, BundleRequestHandler.SUCCESS)
+        except Exception as err:
+            status = BundleRequestHandler.FAILED + f" - {err}"
+            BundleRequestHandler._update_request_status(
+                request_id, status)
+        return request_id
+
+    def get_ipmi_sel_data(self):
+        ipmitool = IpmiFactory().get_implementor('ipmitool')
+        res, err, retcode = ipmitool._run_ipmitool_subcommand("sel list")
+        if retcode != 0:
+            if err.find(ipmitool.VM_ERROR) != -1:
+                err = ("Detected VM environment, can't reach BMC over inband"
+                       "KCS channel, avoiding BMC data collection using IPMI.")
+            logger.warn(
+                f"Unable to fetch SEL list data with an error : {err}")
+        else:
+            with open(self.ipmi_sel_data, 'w+') as sel_list:
+                sel_list.write(res)
+
+    def get_config_data(self):
+        os.makedirs(self.sspl_conf_dir, exist_ok=True)
+        for conf_file in glob.glob('/etc/sspl*.*'):
+            shutil.copy(conf_file, self.sspl_conf_dir)
+
+    def _run_command(self, command):
+        """Run the command and get the response and error returned"""
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+        return_code = process.wait()
+        response, error = process.communicate()
+        return response.rstrip('\n'), error.rstrip('\n'), return_code
+
+    def get_drives_smart_data(self):
```

Comment:
this function can be part of some DriveUtils class, etc

Reply:
Moved it to `DriveUtils` class.